coderd/templateversions.go

```diff
@@ -1040,6 +1041,22 @@ func (api *API) patchActiveTemplateVersion(rw http.ResponseWriter, r *http.Reque
 		})
 		return
 	}
+	job, err := api.Database.GetProvisionerJobByID(ctx, version.JobID)
+	if err != nil {
+		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Internal error fetching template version job status.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+	jobStatus := db2sdk.ProvisionerJobStatus(job)
+	if jobStatus != codersdk.ProvisionerJobSucceeded {
+		httpapi.Write(ctx, rw, http.StatusPreconditionFailed, codersdk.Response{
```

Comment:
This should be 403 Forbidden.

PreconditionFailed refers to [preconditions](https://httpwg.org/specs/rfc9110.html#conditional.requests) that the HTTP client attached to the request in the headers.  We do not use these on our API and so should not return this status.

Note that 403 Forbidden is not just for insufficient permission, it should be used any time the request is well-formed and authentic, but we refuse to process.

Reply:
ha, I saw this code used in one other place that seems similar, but after searching it looks like that was the only usage of it in the whole codebase. :p swapped them all to forbidden.